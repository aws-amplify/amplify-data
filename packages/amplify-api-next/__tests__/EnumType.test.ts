import { expectTypeTestsToPassAsync } from 'jest-tsd';
import { a } from '../index';

// evaluates type defs in corresponding test-d.ts file
it('should not produce static type errors', async () => {
  await expectTypeTestsToPassAsync(__filename);
});

describe('EnumType transform', () => {
  test('Explicit Enum', () => {
    const s = a.schema({
      File: a.model({
        accessLevel: a.ref('AccessLevel'),
      }),
      AccessLevel: a.enum(['public', 'protected', 'private']),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });

  test('Explicit Enum - required', () => {
    const s = a.schema({
      File: a.model({
        accessLevel: a.ref('AccessLevel').required(),
      }),
      AccessLevel: a.enum(['public', 'protected', 'private']),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });

  test('Explicit Enum - auth', () => {
    const s = a.schema({
      File: a.model({
        accessLevel: a.ref('AccessLevel').authorization([a.allow.owner()]),
      }),
      AccessLevel: a.enum(['public', 'protected', 'private']),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });

  test('Implicit Enum', () => {
    const s = a.schema({
      File: a.model({
        accessLevel: a.enum(['public', 'protected', 'private']),
      }),
    });

    const result = s.transform().schema;

    expect(result).toMatchSnapshot();
  });
});
