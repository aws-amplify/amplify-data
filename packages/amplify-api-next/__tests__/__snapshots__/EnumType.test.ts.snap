// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnumType transform Explicit Enum 1`] = `
"type File @model 
{
  accessLevel: AccessLevel
}

enum AccessLevel {
public
protected
private
}"
`;

exports[`EnumType transform Implicit Enum 1`] = `
"type File @model 
{
  accessLevel: AccessLevel
}

enum AccessLevel {
public
protected
private
}"
`;
